Dockerfile

```diff
@@ -8,14 +8,16 @@ ENV BUNDLE_FROZEN="true"
 
 WORKDIR /app
 RUN apk --update --no-cache add git build-base postgresql-dev
-
-COPY . /app
 RUN gem update bundler --no-document
+
+# NB: its more efficient not to copy the full app folder until after the gems are installed (reduces unnecessary rebuilds)
+COPY Gemfile Gemfile.lock .ruby-version /app/
 RUN bundle install --jobs 4 --retry 3 \
      && rm -rf /usr/local/bundle/cache/*.gem \
      && find /usr/local/bundle/gems/ -name "*.c" -delete \
      && find /usr/local/bundle/gems/ -name "*.o" -delete
 
+COPY . /app
```

Comment:
If this is not needed on the build stage, would it make sense to copy it straight into the runtime image?

Reply:
hmm.. not sure... I'll try it and see

Reply:
Hi @mafonso I've tweaked the dockerfile a bit more and delayed copying in the app until right to the end. (We do need to copy in the app folder BTW as all the code is needed - maybe the tests are not, but will leave that for another day.)